pkg/client/client_v2.go

```diff
@@ -529,12 +531,28 @@ func (c *clientV2) syncComponent(expected *proto.CheckinExpected) {
 	if expected.Component == nil {
 		return
 	}
+
 	c.componentMu.Lock()
 	defer c.componentMu.Unlock()
 
-	if expected.Component.Limits.MaxProcs > 0 {
-		_ = runtime.GOMAXPROCS(int(expected.Component.Limits.MaxProcs))
+	// applying the component limits
+	if expected.Component.Limits != nil {
```

Comment:
This logic could have a small unit test, otherwise looks good.

Reply:
Yeah, that's what I've been doing for the last few hours :) This code was buggy, I added tests and fixed the bugs in https://github.com/elastic/elastic-agent-client/pull/83/commits/52a5535723c05c45867ff66b9b5fb75c1e7b4008